builder/support.go

```diff
@@ -1,9 +1,18 @@
 package builder
 
 import (
+	"regexp"
 	"strings"
 )
 
+const acceptableRemoteMIME = `(?:application/(?:(?:x\-)?tar|octet\-stream|((?:x\-)?(?:gzip|bzip2?|xz)))|(?:text/plain))`
+
+var mimeRe = regexp.MustCompile(acceptableRemoteMIME)
+
+func selectAcceptableMIME(ct string) string {
+	return mimeRe.FindString(ct)
```

Comment:
why is this a regex? Are you concerned about types that would contain one of those strings but have additional characters?